src/pages/home/HeaderView.tsx

```diff
@@ -334,8 +348,10 @@ function HeaderView({report, personalDetails, parentReport, parentReportAction,
                                     )}
                                 </PressableWithoutFeedback>
                                 <View style={[styles.reportOptions, styles.flexRow, styles.alignItemsCenter]}>
-                                    {isTaskReport && !isSmallScreenWidth && ReportUtils.isOpenTaskReport(report, parentReportAction) && <TaskHeaderActionButton report={report} />}
-                                    {canJoin && !isSmallScreenWidth && joinButton}
+                                    {isTaskReport && !(isSmallScreenWidth || shouldUseNarrowLayout) && ReportUtils.isOpenTaskReport(report, parentReportAction) && (
```

Comment:
these inline checks are so long and so convoluted. Please extract to separate variable before the `return`, so that JSX stays cleaner